### ISSUE TITLE
[Core] Mortar mapper with linear solver

### ISSUE THREAD

**Issue by rubenzorrilla** [KratosMultiphysics/Kratos#3370]
@loumalouomega 

I tried to use your mortar mapper in the core (`simple_mortar_mapper_process.h`) and it throws this error when specifying the linear solver in the constructor. 
```
Error: attempting to do a GetValue for: NODAL_MAUX variable #2503569152 unfortunately the variable is not in the database and the operations is not threadsafe (this function is being called from within a parallel region)
```
Actually, I don't really know how this works so I can't give you more information than that.

**Comment by loumalouomega** [comment]
I will check, the linear solver is not used by default, I just implemented the first time to create the reference solution. Probably it broke in some of the refactorings I did, I will fix it and add the proper test

**Comment by loumalouomega** [comment]
It is solved in #3376, I will close this to reduce the issues number
